test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs

```diff
@@ -577,5 +577,15 @@ public void It_passes_ridless_target_to_compiler()
                 .Should()
                 .Pass();
         }
+
+        [Fact]
+        public void It_restores_only_ridless_tfm()
```

Comment:
Note this test passes even without this fix because it uses dotnet.exe which already excludes Microsoft.NuGet.targets. But this seems like a good scenario to ensure we don't regress.